src/vma/dev/net_device_val.cpp

```diff
@@ -1051,14 +1056,14 @@ resource_allocation_key* net_device_val::ring_key_redirection_reserve(resource_a
 
 	rings_hash_map_t::iterator ring_iter = m_h_ring_map.begin();
 	int min_ref_count = ring_iter->second.second;
-	resource_allocation_key *min_key = ring_iter->first;
+	resource_allocation_key *min_key = new resource_allocation_key(*ring_iter->first);
 	while (ring_iter != m_h_ring_map.end()) {
 		// redirect only to ring with the same profile
 		if (ring_iter->first->get_ring_profile_key() ==
 		    key->get_ring_profile_key() &&
 		    ring_iter->second.second < min_ref_count) {
 			min_ref_count = ring_iter->second.second;
-			min_key = ring_iter->first;
+			min_key = new resource_allocation_key(*ring_iter->first);
```

Comment:
Is this a memory leak ? why do we need to create it twice ? 
where it will be freed ?